app/javascript/controllers/keywords_controller.js

```diff
@@ -23,16 +23,23 @@ export default class extends Controller {
     this.containerTarget.classList.remove('is-invalid')
   }
 
-  checkForEnter(event) {
-    if (event.key === 'Enter') {
-      event.preventDefault() // prevent form submission
-      this.add(event);
+  checkForDuplicates(event) {
+    const input = event.target.closest('input[type="text"]')
+    const keywords = Array.from(document.querySelectorAll('.keyword-row:not([style*="display: none"]) input[type="text"]'), input => input.value)
```

Comment:
this is the gnarly selector -- basically look for any non-hidden `.keyword-row` elements that have a text box, and map out their values into an array

Reply:
How about extracting this into a hasDuplicateKewords method?  This looks pretty similar to what we have in https://github.com/sul-dlss/argo/blob/6139bc519f915c25cd8e55eccd36a6708b22ef49/app/javascript/controllers/registration_item_row_controller.js#L94-L100